tests/NTest_adc.lua

```diff
@@ -0,0 +1,57 @@
+-- Walk the ADC through a stepped triangle wave using the attached voltage
+-- divider and I2C GPIO expander.
+
+local N = require('NTest')("adc")
+
+N.test('setup', function()
+  -- Configure the ADC
+  if adc.force_init_mode(adc.INIT_ADC)
+  then
+    node.restart()
+    error "Must reboot to get to ADC mode"
+  end
+
+  -- Configure the I2C bus
+  i2c.setup(0, 2, 1, i2c.FAST)
+
+  -- Set the IO expander port B to channels 0 and 1 as outputs
+  i2c.start(0)
+  ok(i2c.address(0, 0x20, i2c.TRANSMITTER))
+  i2c.write(0, 0x01, 0xFC)
+  i2c.stop(0)
+end)
+
+-- set the two-bit voltage divider output value to v (in 0,1,2,3)
+local function setv(v)
+  assert (0 <= v and v <= 3)
+  i2c.start(0)
+  i2c.address(0, 0x20, i2c.TRANSMITTER)
+  i2c.write(0, 0x15, v)
+  i2c.stop(0)
+end
+
+-- read out the ADC and compare to given range
+local function checkadc(min, max)
+  local v = adc.read(0)
+  print("read adc", v, min, max)
+  return v >= min and v <= max
+end
+
+-- Since we have a rail-to-rail 4-tap DAC, as it were, give us some one-sided
+-- wiggle around either rail and some two-sided wiggle around both middle stops
+local vmin = {  0, 300, 700, 1000 }
+local vmax = { 24, 400, 800, 1024 }
+
+-- Set the DAC, wait a brief while for physics, and then read the ADC
+local function mktest(fs, i)
+  N.test(fs:format(i), function()
+    setv(i)
+    print("set v", i)
+    tmr.delay(10)
+    ok(checkadc(vmin[i+1], vmax[i+1]))
```

Comment:
You could add a msg here to show what is failing/succeeding.
You could even move the ok to checkadc, using the string from the print statement there.

``